core/src/main/java/de/jplag/highlightextraction/ContainedMatchesStrategy.java

```diff
@@ -0,0 +1,35 @@
+package de.jplag.highlightextraction;
+
+import static de.jplag.highlightextraction.SubSequenceUtil.addSequence;
+import static de.jplag.highlightextraction.SubSequenceUtil.getSubSequences;
+
+import java.util.List;
+import java.util.Map;
+import java.util.function.Consumer;
+
+import de.jplag.TokenType;
+
+/**
+ * Strategy that uses matches and their submatches, that are longer than minSubSequenceLength from the comparisons and
+ * calculates the frequency of their appearance in full matches across all submissions.
+ */
+public class ContainedMatchesStrategy implements FrequencyStrategy {
+    /**
+     * Adds all submatches with min size length of the matches to a map using the token sequence as the key.
+     * @param matchTokenTypes Token list of the match.
+     * @param frequencyMap Map that contains token subsequences and how often they occur across comparisons.
+     * @param minSubSequenceLength Minimum length of the considered submatches.
+     */
+    @Override
+    public void processMatchTokenTypes(List<TokenType> matchTokenTypes, Map<Integer, Integer> frequencyMap, int minSubSequenceLength) {
+        Consumer<List<TokenType>> sequenceConsumer = seq -> addSequence(frequencyMap, seq);
+        List<List<TokenType>> subSequences = getSubSequences(matchTokenTypes, minSubSequenceLength);
+        for (List<TokenType> subSequence : subSequences) {
+            frequencyMap.putIfAbsent(subSequence.hashCode(), 0);
```

Comment:
After a discussion with @TwoOfTwelve: We should probably use the list itself as key instead of the hashcode to reduce the chance of collisions. This means we just replace `subSequence.hashCode()` with `subSequence` and also change the hashmap to `Map<List<TokenType>, Integer>`.